### ISSUE TITLE
aws-load-balancer-controller charts missing certificate issuer

### ISSUE THREAD

**Issue by kuldazbraslav** [kubernetes/kops#12981]
/kind bug

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
`Version 1.21.4 (git-53e6bf3e5b0a77c78df2a7c60baca6e926fd5105)` , however according to the charts it's not fixed in higher versions either

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.20", GitCommit:"1f3e19b7beb1cc0110255668c4238ed63dadb7ad", GitTreeState:"clean", BuildDate:"2021-06-16T12:58:51Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.20", GitCommit:"1f3e19b7beb1cc0110255668c4238ed63dadb7ad", GitTreeState:"clean", BuildDate:"2021-06-16T12:51:17Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**3. What cloud provider are you using?**
AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
kops create cluster with spec containing
```
awsLoadBalancerController:
  enabled: true
```

**5. What happened after the commands executed?**
All the resources are created but the pod won't start due to missing issuer for the certificate. The issuer is defined in the [parent charts](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/docs/install/v2_2_0_full.yaml) but missing in [kops](https://github.com/kubernetes/kops/blob/v1.21.4/upup/models/cloudup/resources/addons/aws-load-balancer-controller.addons.k8s.io/k8s-1.9.yaml.template)

**6. What did you expect to happen?**
aws-load-balancer-controller to start up

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

not cluster-spec related, yet attached: https://gist.github.com/kuldazbraslav/b5d059b785313f6cc081a8fc0cfd9b89

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

**9. Anything else do we need to know?**

**Comment by olemarkus** [comment]
Can you post your full cluster spec?

**Comment by kuldazbraslav** [comment]
Added to the original post. However this issue is clearly not cluster spec related, the Issuer object is missing in the kops charts.

**Comment by olemarkus** [comment]
There are no charts in kops. kops does not use helm.

The issuer is installed by kops itself when the addon is enabled. The cluster spec providers the values passed into the addons, but is also needed for us to reproduce the error.

Especially in your case where I see the cluster spec is heavily modified.

**Comment by kuldazbraslav** [comment]
So if I understand it right, kops takes the addons (which I refer to as charts, not helm) from S3 bucket where kops uploads them. The source code of the charts seems to be located here https://github.com/kubernetes/kops/blob/master/upup/models/cloudup/resources/addons/aws-load-balancer-controller.addons.k8s.io/k8s-1.9.yaml.template .

The Certificate object (starting on line 789 refers to issuer `aws-load-balancer-controller.addons.k8s.io`) which is not in the file (while it is in https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/download/v2.3.1/v2_3_1_full.yaml) which is referred in the comment at the top of aws-load-balancer-controller.addons.k8s.io/k8s-1.9.yaml.template .

When I deployed the issuer manually, it fixed the issue.
```
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: aws-load-balancer-controller.addons.k8s.io
  namespace: kube-system
spec:
  selfSigned: {}
```

**Comment by kuldazbraslav** [comment]
Actually I was just curious if the Issuer should be created by something else than the addon template. If it's as easy as adding the issuer to the template, I can get a PR ready.

**Comment by olemarkus** [comment]
The issuer is created by the channels binary.
Do you have any outstanding updates? E.g is the control plane nodes running with kops-controller:1.21 or above?

**Comment by kuldazbraslav** [comment]
They are running the newest kops-controller. However I was upgrading from previous version of kops, so let me test the whole upgrade path again, I'll keep you posted.

**Comment by kuldazbraslav** [comment]
So it seems that the problem is in the upgrade path:

1. kops update cluster is run
2. that deploys both aws-load-balancer-controller (failing to start due to missing secret -> not created due to missing issuer) and kops-controller (not scheduled to any nodes due to missing label `kops.k8s.io/kops-controller-pki: ""` on control plane nodes)
3. without touching the issuer, rolling update can't be performed - cluster won't validate
4. Removing the node label from kops-controller won't help either, the control plane nodes are missing `/etc/kubernetes/kops-controller/` host path

In the previous round I checked it after manually creating the issuer and running rolling update (which deployed the proper node labels and thus kops controller got properly scheduled).

Thanks for your help, this seems related only to my upgrade path where I'd like to skip multiple rolling updates by skipping more kops versions. Closing.
